book/src/qobject/generated-qobject.md

```diff
@@ -0,0 +1,111 @@
+As mentioned before, the C++ QObject is exposed to Rust as an opaque CXX type.
+This allows you to implement methods for it in Rust using normal `impl` blocks.
+Because `qobject::T` is an opaque C++ type, the same rules as with normal CXX opaque types apply.
+Most importantly this means that the type may never be accessed by value or by mutable self reference directly.
+Rather, all methods must use either `&self` or `self: Pin<&mut Self>` as the self types. This prevents Rust from moving the data in memory, which would invalidate C++ pointers to it.
+
+For more information about pinning, refer to the [pin documentation](https://doc.rust-lang.org/std/pin/). 
+
+In addition to methods that extend the Rust interface of a `qobject::T`, you may also mark methods within an `impl qobject::T` block with the `#[qinvokable]` attribute.
+These methods will be exposed to the C++ QObject and can be called by C++ or QML.
+See the [QObject page](./qobject_struct.md#invokables) for more details.
+
+## Available Functions
+
+Every `qobject::T` struct provides the following methods:
+
+### Access to the Qt thread
+``` rust,ignore,noplayground
+fn qt_thread(&self) -> UniquePtr<CxxQtThread>
+```
+This function provides you with a handle to the Qt thread that the QObject resides in.
+This is helpful as the QObject itself does not implement [Send](https://doc.rust-lang.org/std/marker/trait.Send.html) nor [Sync](https://doc.rust-lang.org/std/marker/trait.Sync.html).
+The CxxQtThread however is Send and can therefore be moved into a different thread.
+By using the `queue` function on the CxxQtThread, you may then queue a Rust lambda onto the Qt thread again.
+The lambda also takes a pinned mutable reference to the QObject, so that it can modify it.
+
+See the [CxxQtThread page](./cxxqtthread.md) for more details.
+
+### Signal emission
+``` rust,ignore,noplayground
+fn emit(self: Pin<&mut Self>, signal: /*Your Signals enum goes here*/)
+```
+If there is a [Signals enum](./signals_enum.md) defined, CXX-Qt will generate the appropriate `emit` function to allow you to emit signals.
+
+See the [Signals enum page](./signals_enum.md) for more details.
+
+### Access to internal Rust struct
+For every field in the Rust struct, CXX-Qt will generate appropriate getters and setters for every field in the Rust struct.
+See the [QObject page](./qobject_struct.md#properties) for details.
+
+There is also an advanced way to access the data in the internal Rust struct:
+``` rust,ignore,noplayground
+fn rust(&self) -> &T
+unsafe fn rust_mut(self: Pin<&mut Self>) -> &mut T
+```
+Where `T` is the struct with the `#[cxx_qt::qobject]` macro.
+
+This allows you to directly manipulate the internal Rust struct without having to use the generated accessor methods.
+
+You may notice that the mutable version of this is `unsafe` to call.
+This is by design.
+Modifying a field that corresponds to a `#[qproperty]` without calling the appropriate changed signal may cause undefined behavior.
```

Comment:
Would this really be undefined behavior? Or merely a logic error?

Reply:
Undefined on the QML/Qt side :-) If you don't call a changed signal then things won't update in QML, but other items which happen to trigger a binding update or call an invokable will have the new values. So then you get this mismatch of state within the UI of your application.

I've seen very odd things happen with this in real world QML apps, hence undefined behaviour.

Reply:
"Undefined behavior" has a more specific meaning than "very odd things happen". Very odd things usually do happen with undefined behavior, but more specifically it means the compiler and CPU could do *anything* and you shouldn't rely on whatever a particular implementation happens to do.

Reply:
OK, changed to "a logic error"

Reply:
Hmm, this still feels weird. How about: "Modifying a field that corresponds to a `#[qproperty]` without calling the appropriate changed signal may violate invariants of C++ code."

Reply:
Changed to "a logic error in C++/QML code"